module5/lessons/Week3/Authentication.md

```diff
@@ -0,0 +1,289 @@
+
+## Implementing Identity
+
+Now that we have seen a 'handrolled' authentication solution, let's take a look at using [ASP.NET Core Identity](https://learn.microsoft.com/en-us/aspnet/core/security/authentication/identity?view=aspnetcore-6.0&tabs=visual-studio)
+
+From the documentation linked above:
+
+> ASP.NET Core Identity:
+> * Is an API that supports user interface (UI) login functionality.
+> * Manages users, passwords, profile data, roles, claims, tokens, email confirmation, and more.
+
+
+ASP.NET Core Identity is not the _only_ way that we can implement authentication; we have seen previously that we can create our own authentication functionality.  But, most modern web applications use some framework (like Identity) to help manage their authentication.  So, it is important to have worked with one of these frameworks - this will give you an understanding of the possible complexities associated with authentication (and authorization).
+
+### Setup for Identity
+
+We are going to start with a user-free [MessageBoard](https://github.com/turingschool-examples/IdentityMessageBoard) and add authentication to it using ASP.NET Core Identity. Make sure you **fork** and clone the starter!
+
+<section class='instructor-notes' markdown='1'>
+
+The completed code for this lesson can be found at the [authentication-finished](https://github.com/turingschool-examples/IdentityMessageBoard/tree/authentication-finished)
+
+</section>
+
+### Prep Context and Models
+
+In order to use the Identity API, we need to configure our Database Context to build Identity related tables.
+
+* Install the nuget package `Microsoft.AspNetCore.Identity.EntityFramework version 6.0.23`
+* In **DataAccess/MessageBoardContext.cs** update your inheritance to match this:
+    ```c#
+    using IdentityMessageBoard.Models;
+    using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
+    using Microsoft.EntityFrameworkCore;
+
+    namespace IdentityMessageBoard.DataAccess
+    {
+        public class MessageBoardContext : IdentityDbContext
+        {
+            public DbSet<Message> Messages { get; set; }
+
+            public MessageBoardContext(DbContextOptions<MessageBoardContext> options) : base(options)
+            {
+
+            }
+        }
+    }
+    ```
+### Add Custom User Properties
+
+Out of the box, we get a User from ASP.NET Core Identity.  In our database, we can see a table called `AspNetUsers` which holds records that are associated with a class called `IdentityUser`.  This class is defined in the ASP.NET Core Identity package, so we don't have easy access to update it to include information specific to our application.
+
+So, we will use **Inheritance** to create a customer user class that we will reference throughout our application.
+
+```c#
+
+using Microsoft.AspNetCore.Identity;
+
+namespace IdentityMessageBoard.Models
+{
+    public class ApplicationUser : IdentityUser
+    {
+
+    }
+}
+```
```

Comment:
Do we need to make a migration and update the database somewhere around here to get this user set up?

Reply:
lol needed to read further, ignore this.